src/modules/launcher/PowerLauncher/ViewModel/ResultsViewModel.cs

```diff
@@ -236,7 +236,6 @@ public void AddResults(List<Result> newRawResults, string resultId, Cancellation
                 ct.ThrowIfCancellationRequested();
             }
 
-            Results.RemoveAll(r => r.Result.PluginID == resultId);

```

Comment:
Why was the metadata arguement and this statement removed? I was able to use that for the second indexer query to make sure indexer results don't get duplicated.

Reply:
Rather than removing each plugin result individually, I am removing them together in MainViewModel. This is to prevent cases where `var plugins = PluginManager.ValidPluginsForQuery(query);` doesn't return a plugin whose results are still present in the list view.  For example for query 'C:\', PluginManager will not return shell plugin and its results won't be cleared. 

And this function is still present in ResultsViewModel in case you want to use it like this.